healsparse/io_map_healpix.py

```diff
@@ -0,0 +1,52 @@
+import os
+import numpy as np
+
+from .fits_shim import _make_header, _write_healpix_filename
+
+
+def _write_map_healpix(hsp_map, filename, clobber=False):
+    """
+    Internal method to write a HealSparseMap with healpix EXPLICIT format.
+
+    Note that the coverage map is not persisted in this format, and only
+    floating point value maps are supported.
+
+    Parameters
+    ----------
+    hsp_map : `HealSparseMap`
+        HealSparseMap to write to a file.
+    filename : `str`
+        Name of file to save
+    clobber : `bool`, optional
+        Clobber existing file?  Default is False.
+
+    Raises
+    ------
+    RuntimeError if file exists and clobber is False.
+    NotImplementedError if persisting anything other than a floating point map.
+    """
+
+    if hsp_map.is_rec_array:
+        raise NotImplementedError("The healpix EXPLICIT output format is not supported for recarray maps.")
+    if hsp_map.is_integer_map:
+        raise NotImplementedError("The healpix EXPLICIT output format is not supported for integer maps.")
+
+    if os.path.isfile(filename) and not clobber:
+        raise RuntimeError("Filename %s exists and clobber is False." % (filename))
+
+    hdr = _make_header(hsp_map.metadata)
+    hdr['PIXTYPE'] = 'HEALPIX'
+    hdr['INDXSCHM'] = 'EXPLICIT'
+    hdr['ORDERING'] = 'NESTED'
+    hdr['NSIDE'] = hsp_map._nside_sparse
```

Comment:
An `OBS_NPIX` header is documented as "required" in the HEALPix-FITS-format PDF referenced in this PR, as the "number of pixels listed in this file".  Of course, this is redundant with information already in the FITS binary table, but I think it should be provided.